site/gulpfile.js

```diff
@@ -1,137 +1,163 @@
-    const downloadList = swagList.map(s => ({
-        url: s.image,
-        file: escapeName(s.image) + '.jpg',
-    }));
-    return download(downloadList)
+    return download(swagImages)
         .pipe(gulp.dest('dist/assets/swag-img'));
 });
 
 gulp.task('swag-img:optimize', () => {
     return gulp.src('dist/assets/swag-img/*')
-        .pipe(responsive({
-            '**/*': {
-                height: 300,
-                format: 'jpeg',
-                flatten: true,
-            },
-        }, {
-            quality: 90,
-            progressive: true,
-            compressionLevel: 9,
-            errorOnEnlargement: false,
-            errorOnUnusedConfig: false,
-        }))
+        .pipe(responsive([{
+            name: '**/*',
+            height: 300,
+            format: 'jpeg',
+            flatten: true
+        }], RESIZE_OPTS))
         .pipe(gulp.dest('dist/assets/swag-img'));
 });
 
-gulp.task('swag-img:clean', () => {
-    return del('dist/assets/swag-img/**/*');
-});
+gulp.task('swag-img', gulp.series('swag-img:clean', 'swag-img:download' , 'swag-img:optimize'));
 
-gulp.task('swag-img:build-data', (cb) => {
-    return mkdirp('dist/assets', () => {
-        writeFile('dist/assets/data.json', JSON.stringify(builtSwagList), cb);
-    });
+gulp.task('clean', () => {
+    return del([
+        './rev-manifest.json',
+        'dist/assets/css/*',
+        'dist/assets/js/*',
+        'dist/assets/swag-img/*'
+    ]);
 });
 
-gulp.task('swag-img', gulp.parallel('swag-img:build-data', gulp.series('swag-img:clean', 'swag-img:download' , 'swag-img:optimize')));
+gulp.task('cachebust', cb => {
+    const bustedFiles = [
+        'dist/assets/css/*',
+        'dist/assets/js/*',
+        'dist/assets/swag-img/*'
+    ];
+
+    return gulp.src(bustedFiles, {base: 'dist/assets/'})
+        .pipe(cachebust())
+        .pipe(cacheClean())
+        .pipe(gulp.dest('dist/assets'))
+        .pipe(cachebust.manifest({
+            base: 'dist/assets',
+            merge: true
+        }))
+        .pipe(gulp.dest('dist/assets/'))
+        .on('end', () => {
+            delete require.cache[require.resolve('./rev-manifest.json')];
+            const manifest = require('./rev-manifest.json');
+            swagList.forEach(swag => {
+                const filename = `swag-img/${swag.image.split('/').pop()}`;
+                if (!manifest[filename]) {
+                    console.warn('Unable to find image in manifest:', filename);
+                    return;
+                }
+                swag.image = `/assets/${manifest[filename]}`;
+            });
+
+            cb();
+        });
+});
 
-gulp.task('build', gulp.parallel('pug', 'styl', 'js', 'img', 'swag-img'));
+gulp.task('webserver', () => {
+    return gulp.src('dist')
+        .pipe(webserver({
+            livereload: true,
+            open: true
+        }));
+});
 
 gulp.task('watch', () => {
     gulp.watch('src/pug/**/*.pug', gulp.parallel('pug'));
```

Comment:
Here again, :star:s

Sorry for the flood :fearful:

Reply:
This would need to be recurisve, but not js; if you change includes/body, pug needs to rebuild